ts/components/ChooserListSearch.tsx

```diff
@@ -1,27 +1,28 @@
+import { Option } from "fp-ts/lib/Option";
 import I18n from "i18n-js";
 import * as pot from "italia-ts-commons/lib/pot";
 import { Text, View } from "native-base";
-import React from "react";
-import {
-  Image,
-  ImageSourcePropType,
-  ListRenderItem,
-  StyleSheet
-} from "react-native";
+import React, { ComponentProps } from "react";
+import { Image, ImageSourcePropType, StyleSheet } from "react-native";
 import customVariables from "../theme/variables";
 import ChooserList from "./ChooserList";
+import ChooserListItem from "./ChooserListItem";
 
 type OwnProps<T> = {
   listState: ReadonlyArray<T>;
   searchText: string;
-  keyExtractor: (item: T, index: number) => string;
-  renderItem: ListRenderItem<T>;
   onSearchItemContainsText?: (item: T, searchText: string) => boolean;
   noSearchResultsSourceIcon?: ImageSourcePropType;
   noSearchResultsSubtitle?: string;
+  keyExtractor: (item: T) => string;
+  itemTitleExtractor: (item: T) => string;
+  selectedItemIds: Option<Set<string>>;
 };
 
-type Props<T> = OwnProps<T>;
+type ChooserListItemProps = "itemIconComponent" | "onPressItem";
```

Comment:
what does it mean?
Could you add an explanation comment?

Reply:
Are the properties that this component will have to pass to the ChooserListItem component and so I defined them the set of properties required and defined under `Pick<ComponentProps<typeof ChooserListItem>, ChooserListItemProps>`